nvflare/app_common/statistics/numeric_stats.py

```diff
@@ -160,14 +156,18 @@ def accumulate_hists(
             if feature not in global_hists:
                 g_bins = []
                 for bucket in hist.bins:
-                    g_bins.append(Bin(bucket.low_value, bucket.high_value, bucket.sample_count))
+                    g_bins.append(
+                        Bin(
+                            round(bucket.low_value, precision), round(bucket.high_value, precision), bucket.sample_count
+                        )
+                    )
                 g_hist = Histogram(HistogramType.STANDARD, g_bins)
                 global_hists[ds_name][feature] = g_hist
             else:
                 g_hist = global_hists[ds_name][feature]
-                g_buckets = bins_to_dict(g_hist.bins)
+                g_buckets = bins_to_dict(g_hist.bins, precision)
```

Comment:
when the bin is created, it is already rounded though, should the `bins_to_dict` round it one more time?

I think `bins_to_dict` should just convert one structure to another, so we should not add round in there.